internal/injector/injector.go

```diff
@@ -87,44 +91,64 @@ func New(pkgDir string, opts Options) (*Injector, error) {
 
 	var (
 		pkgPath     string
-		dec         *decorator.Decorator
+		decorators  []*decorator.Decorator
 		restorerMap map[string]string
 	)
 	if pkgs, err := decorator.Load(cfg, pkgDir); err != nil {
 		return nil, err
 	} else {
-		pkg := pkgs[0]
-		switch len(pkg.Errors) {
-		case 0:
-			// Nothing to do, this is a success!
-		case 1:
-			return nil, pkg.Errors[0]
-		default:
-			return nil, fmt.Errorf("%w (and %d more)", pkg.Errors[0], len(pkg.Errors)-1)
-		}
-
-		dec = pkg.Decorator
-		pkgPath = pkg.PkgPath
-		restorerMap = make(map[string]string, len(pkg.Imports)+len(builtin.RestorerMap))
+		decorators = make([]*decorator.Decorator, 0, len(pkgs))
+		restorerMap = make(map[string]string, len(builtin.RestorerMap))
 		for path, name := range builtin.RestorerMap {
 			restorerMap[path] = name
 		}
-		for _, imp := range pkg.Imports {
-			if imp.Name == "" {
-				// Happens when there is an error while processing the import, typically inability to resolve the name due to a
-				// typo or something. If we allow blank names in the map, the restorer just removes the qualifiers, which is
-				// obviously undesirable.
-				continue
+
+		for _, pkg := range pkgs {
+			switch len(pkg.Errors) {
+			case 0:
+				// Nothing to do, this is a success!
+			case 1:
+				return nil, pkg.Errors[0]
+			default:
+				return nil, fmt.Errorf("%w (and %d more)", pkg.Errors[0], len(pkg.Errors)-1)
```

Comment:
```suggestion
				return nil, fmt.Errorf("%d errors occured: %w", len(pkg.Errors), errors.Join(pkg.Errors...))
```

Reply:
Oh you spotted a leftover form before the "go1.21 or bust" era :)